### DISCUSSION TITLE
Does plugin:@next/next/recommended include all default next lint tests

### DISCUSSION THREAD

**Discussion by AndrewPark9517** [vercel/next.js#72351] [Help]
### Summary

Am looking to disable the lint test during build using the config outlined [here](https://nextjs.org/docs/app/api-reference/next-config-js/eslint) because there is a separate eslint step in our workflow.

The [documentation](https://nextjs.org/docs/pages/building-your-application/configuring/eslint#recommended-plugin-ruleset) suggests using `@next/next/recommended` for my case, but am uncertain what this plugin actually includes. 
Does this plugin include all the base tests that are included in the default lint step in build?

**Comment by crossterra-agent** [comment]
Waiting for developer to complete security fix on PR #1984 (credential scoping — DB secrets must only be forwarded when `role == data-reader`). Will invoke reviewer once developer reports success.

PR: https://github.com/CrossTerra/monorepo/pull/1984

**Comment by qinacepuc31-max** [comment]
nformiere dich vorab über verfügbare Spiele, Angebote, Teilnahmebedingungen und geltende Vorschriften. Nur für Erwachsene ab dem gesetzlichen Mindestalter. Glücksspiel kann süchtig machen – spiele https://greatwin.life/ verantwortungsvoll, setze dir klare Grenzen und behalte deine Ausgaben im Blick.

**Comment by hamedrabah** [comment]
Short answer: **no**. `plugin:@next/next/recommended` is the recommended ruleset from the Next.js plugin only; it is not the complete configuration that the old default `next lint` / build lint step used.

For a Next 14/15 TypeScript app, the closest equivalent to the create-next-app defaults is to run ESLint separately with the full Next configs:

```js
// .eslintrc.js (legacy config used by Next 14/early 15)
module.exports = {
  extends: ['next/core-web-vitals', 'next/typescript'],
}
```

`eslint-config-next` supplies React, React Hooks, and Next.js rules; `core-web-vitals` promotes the performance-related Next rules, and `next/typescript` adds the TypeScript recommended rules. Extending only `plugin:@next/next/recommended` is appropriate when another config already supplies the parser and React/import/a11y rules, but it will not reproduce all of the old defaults.

There is also an important version boundary now:

- **Next 15 and earlier:** `eslint.ignoreDuringBuilds: true` skips the build-time lint step, so the separate CI ESLint command needs the full config above.
- **Next 16:** `next build` no longer runs linting, `next lint` and the `eslint` option were removed, and the supported path is the ESLint CLI with flat config (for example `eslint-config-next/core-web-vitals`).

Official references:
- https://nextjs.org/docs/15/app/api-reference/config/eslint
- https://nextjs.org/docs/app/api-reference/config/eslint
- https://nextjs.org/docs/app/guides/upgrading/version-16
